packages/components/src/components/LogoutButton/LogoutButton.js

```diff
@@ -18,35 +18,36 @@ import React, { Component } from 'react';
 import './LogoutButton.scss';
 
 /* istanbul ignore next */
-function handleLogout() {
-  window.location.href = '/oauth/sign_out';
+function handleLogout(url) {
+  window.location.href = url;
 }
 
 export class LogoutButton extends Component {
   state = {
-    showLogout: false
+    logoutUrl: null
   };
 
   componentDidMount() {
-    this.determineOpenShift();
+    this.determineLogoutUrl();
   }
 
-  async determineOpenShift() {
+  async determineLogoutUrl() {
     try {
-      const showLogout = await this.props.shouldDisplayLogout();
-      this.setState({ showLogout });
+      const logoutUrl = await this.props.getLogoutUrl();
+      this.setState({ logoutUrl });
     } catch (error) {} // eslint-disable-line
   }
 
   render() {
-    if (!this.state.showLogout) {
+    const { logoutUrl } = this.state;
+    if (!logoutUrl) {
       return null;
     }
     return (
       <HeaderGlobalAction
         data-testid="logout-btn"
         className="tkn--logout-btn"
-        onClick={handleLogout}
+        onClick={() => handleLogout(logoutUrl)}
```

Comment:
we can define `handleLogout` on the LogoutButton class and avoid creating a new anonymous function on each render here:

`onClick={this.handleLogout}`

and

```js
handleLogout = () => {
  const { logoutURL } = this.state;
  window.location.href = logoutURL;
};

render() {
...
```